keras_cv/layers/regularization/dropblock_2d.py

```diff
@@ -0,0 +1,195 @@
+            ordering of the dimensions in the inputs. channels_last corresponds to
+            inputs with shape (batch_size, height, width, channels) while channels_first
+            corresponds to inputs with shape (batch_size, channels,height, width). It
+            defaults to the image_data_format value found in your Keras config file at
+            ~/.keras/keras.json. If you never set it, then it will be channels_last.
+        seed: integer. To use as random seed.
+        name: string. The name of the layer.
+
+    Usage:
+    DropBlock2D can be used inside a `tf.keras.Model`:
+    ```python
+    # (...)
+    x = Conv2D(32, (1, 1))(x)
+    x = BatchNormalization()(x)
+    x = ReLU()(x)
+    x = DropBlock2D()(x)
+    # (...)
+    ```
+    When used directly, the layer will zero-out some inputs in a contiguous region and
+    normalize the remaining values.
+
+    ```python
+    # Small feature map shape for demonstration purposes:
+    features = tf.random.stateless_uniform((1, 4, 4, 1), seed=[0, 1])
+
+    # Preview the feature map
+    print(features[..., 0])
+    # tf.Tensor(
+    # [[[0.08216608 0.40928006 0.39318466 0.3162533 ]
+    #   [0.34717774 0.73199546 0.56369007 0.9769211 ]
+    #   [0.55243933 0.13101244 0.2941643  0.5130266 ]
+    #   [0.38977218 0.80855536 0.6040567  0.10502195]]], shape=(1, 4, 4),
+    # dtype=float32)
+
+    layer = DropBlock2D(dropblock_size=2, seed=1234)  # Small size for demonstration
+    output = layer(features, training=True)
+
+    # Preview the feature map after dropblock:
+    print(output[..., 0])
+    # tf.Tensor(
+    # [[[0.10955477 0.54570675 0.5242462  0.42167106]
+    #   [0.46290365 0.97599393 0.75158674 1.3025614 ]
+    #   [0.         0.         0.39221907 0.6840355 ]
+    #   [0.         0.         0.80540895 0.14002927]]], shape=(1, 4, 4),
+    # dtype=float32)
+
+    ```
+    We can observe two things:
+    1. A 2x2 block has been set to zero.
+    2. The inputs have been normalized.
+
+    One must remember, that DropBlock operation is random, so bigger or smaller
+    patches can be dropped.
+    """
+
+    def __init__(
+        self,
+        keep_probability=0.9,
+        dropblock_size=7,
+        data_format=None,
+        seed=None,
+        name=None,
+    ):
+        super().__init__(seed=seed, name=name, force_generator=True)
+        if not dropblock_size > 0:
+            raise ValueError(
+                f"dropblock_size must be greater than 0. Received: {dropblock_size}"
+            )
+        if not 0.0 <= keep_probability <= 1.0:
+            raise ValueError(
+                f"keep_probability must be a number between 0 and 1. "
+                f"Received: {keep_probability}"
+            )
+
+        self._keep_probability = keep_probability
+        self._dropblock_size = dropblock_size
+        self._data_format = conv_utils.normalize_data_format(data_format)
+
+    def call(self, x, training=False):
```

Comment:
The default value of "training" should usually be None (for a normal layer), or True (if it is a preprocessing layer)

Reply:
Should there also be:
```python
if training is None:
    training = tf.keras.backend.learning_phase()
```
?

Reply:
The learning_phase() is mostly for tf v1 behavior, which we can skip here. You can assume training is boolean here.

Reply:
Changed `training=None`